src/lib.rs

```diff
@@ -174,7 +176,11 @@ impl GpuProfiler {
 
         // Map all buffers.
         for pool in self.active_frame.query_pools.iter_mut() {
-            pool.buffer_mapping = Some(pool.resolved_buffer_slice().map_async(wgpu::MapMode::Read).boxed());
+            let mapped_buffers = self.active_frame.mapped_buffers.clone();
+            pool.resolved_buffer_slice().map_async(wgpu::MapMode::Read, move |res| {
+                res.unwrap();
+                mapped_buffers.fetch_add(1, std::sync::atomic::Ordering::SeqCst);
```

Comment:
SeqCst is a bit too much here. I'm quite sure Ordering::Release is the correct ordering semantic here: We want to ensure that anyone seeing this variable updated (i.e. potentially hitting `frame.query_pools.len()`) will see any other writes (i.e. the buffer being mapped)

pondering actually if relaxed might be enough since pending_frames can't be access in parallel while we're in here, meaning frame.query_pools.len() needs no syncing and the semantics of map should be enough to ensure there is a barrier..... but this gives me a headache so let's not go there ;)